classes/modules/turnitin_quiz.class.php

```diff
@@ -87,7 +87,8 @@ public function update_mark($attemptid, $identifier, $userid, $grade, $quizgrade
         foreach ($attempt->get_slots() as $slot) {
             $answer = $attempt->get_question_attempt($slot)->get_response_summary();
             // Check if this is the slot the mark is for by matching content.
-            if (sha1($answer.$slot) == $identifier) {
+            $answerslot = $answer ? $answer.$slot : $answer;
```

Comment:
If $answer, when coerced to a boolean, is false, it seems like we should use `$slot` on its own.

```suggestion
            $answerslot = $answer ? $answer.$slot : $slot;
```